src/middleware.ts

```diff
@@ -122,13 +122,52 @@ type StateStorage = {
   setItem: (name: string, value: string) => void | Promise<void>
 }
 type PersistOptions<S> = {
+  /** Name of the storage (must be unique) */
   name: string
-  storage?: StateStorage
+  /**
+   * A function returning a storage.
+   * The storage must fit `window.localStorage`'s api.
```

Comment:
More precisely, it's not localStorage api because it also supports async version of it.

Reply:
True, what about: "The storage must fit `window.localStorage`'s api (or an async version of it)." ?